user-interface/src/main/java/life/qbic/datamanager/views/projects/project/samples/SampleContentComponent.java

```diff
@@ -308,6 +310,16 @@ private void editBatch(EditBatchDialog.ConfirmEvent confirmEvent) {
     });
   }
 
+  private void removeRowFromBatch(EditBatchDialog.RemoveRowEvent removeRowEvent) {
+    EditBatchDialog dialog = removeRowEvent.getSource();
+    if(deletionService.isSampleRemovable(removeRowEvent.getSampleInfo().getSampleId(),
+        context.projectId().orElseThrow())) {
```

Comment:
I find it hard to read this statement. Extracting 2 variables would make it easier to read (IntelliJ can do that for you with cmd + option + v).